daffodil-test/src/test/scala/org/apache/daffodil/section05/facets/TestNulChars.scala

```diff
@@ -35,6 +35,5 @@ class TestNulChars {
   // DAFFODIL-2363 &#xE000; (NUL replacement into XML) can't be used in pattern facet. With full validation.
   @Test def test_nulPattern1() = { runner.runOneTest("nulPattern1") }
 
-  // DAFFODIL-2364 - infinite loop
-  // @Test def test_nulPad1() = { runner.runOneTest("nulPad1") }
+  @Test def test_nulPad1() = { runner.runOneTest("nulPad1") }
```

Comment:
Unless there already is one, I suggest we need another test which is the real "sparse-array" situation:
I.e., the input data would be ```00 00 31 00 00 32 00 00 33 00 00``` and the array will have 3 children, the absent ones because of the nul chars would be Absent and so not added to the infoset.